lib/pbench/test/unit/server/query_apis/test_datasets_owner.py

```diff
@@ -0,0 +1,249 @@
+        create_drb_user,
+    ):
+        """
+        Check behavior of the change dataset_owner API with various combinations of dataset
+        owner (managed by the "owner" parametrization here) and authenticated
+        user (managed by the build_auth_header fixture).
+        """
+        self.fake_elastic(monkeypatch, get_document_map, False)
+
+        is_admin = build_auth_header["header_param"] == HeaderTypes.VALID_ADMIN
+        if not HeaderTypes.is_valid(build_auth_header["header_param"]):
+            expected_status = HTTPStatus.UNAUTHORIZED
+        elif not is_admin:
+            expected_status = HTTPStatus.FORBIDDEN
+        else:
+            expected_status = HTTPStatus.OK
+
+        ds = Dataset.query(name=owner)
+        assert_id = str(create_drb_user.id)
+        response = client.post(
+            f"{server_config.rest_uri}/datasets/change_owner/{ds.resource_id}",
+            headers=build_auth_header["header"],
+            query_string={"owner": create_drb_user.username},
+        )
+
+        assert response.status_code == expected_status
+        if expected_status == HTTPStatus.OK:
+            assert response.json == {"ok": 31, "failure": 0}
+            dataset = Dataset.query(name=owner)
+            assert dataset.owner_id == assert_id
+
+    def test_no_dataset(
+        self,
+        client,
+        get_document_map,
+        monkeypatch,
+        pbench_admin_token,
+        server_config,
+        create_drb_user,
+    ):
+        """
+        Check the change_ownership API if the dataset doesn't exist.
+        """
+
+        response = client.post(
+            f"{server_config.rest_uri}/datasets/change_owner/badwolf",
+            headers={"authorization": f"Bearer {pbench_admin_token}"},
+            query_string={"owner": str(create_drb_user.username)},
+        )
+
+        # Verify the report and status
+        assert response.status_code == HTTPStatus.NOT_FOUND
+        assert response.json["message"] == "Dataset 'badwolf' not found"
+
+    def test_no_index(
+        self,
+        client,
+        monkeypatch,
+        attach_dataset,
+        pbench_admin_token,
+        server_config,
+        create_drb_user,
+    ):
+        """
+        Check the change_ownership API if the dataset has no INDEX_MAP. It should
+        fail with a CONFLICT error.
+        """
+        self.fake_elastic(monkeypatch, {}, True)
+
+        ds = Dataset.query(name="drb")
+        response = client.post(
+            f"{server_config.rest_uri}/datasets/change_owner/{ds.resource_id}",
+            headers={"authorization": f"Bearer {pbench_admin_token}"},
+            query_string={"owner": str(create_drb_user.username)},
+        )
+
+        # Verify the report and status
+        assert response.status_code == HTTPStatus.CONFLICT
+        assert response.json == {
+            "message": "Dataset update requires 'Indexed' dataset but state is 'Indexed'"
+        }
+
+    def test_exception(
```

Comment:
I think we can refactor this test in our `commons.py` and let it skip for all the non-elasticsearch tests. (We don't have to tackle that in this PR though)